engines/suppliers/app/models/suppliers/orders_frequency.rb

```diff
@@ -0,0 +1,32 @@
+require 'ice_cube'
+
+module Suppliers
+  class OrdersFrequency < ActiveRecord::Base
+    self.table_name = :orders_frequencies
+
+    FREQUENCY_TYPES = { confirmation: 0, delivery: 1 }.freeze
+
+    serialize :frequency, IceCube::Schedule
+
+    belongs_to :group
+
+    validates :group, :frequency, :frequency_type, presence: true
+    validates :frequency_type, inclusion: { in: FREQUENCY_TYPES.values }
+    validates :frequency_type, uniqueness: { scope: :group_id }
+
+    delegate :to_ical, to: :frequency
+
+    attr_accessor :ical
```

Comment:
instead of having this tied to the before validation, IMO it's simpler an easier to understand if we have the setter deal with the frequency. That's what you would do in a PORO:

```ruby
def ical=(value)
  self.frequency = IceCube::Schedule.from_ical(value)
end
```

And you could 🔥  the before_validation, although you would need to add a validation